.github/workflows/aws-deploy.yaml

```diff
@@ -12,70 +12,26 @@ jobs:
     steps:
       - uses: actions/checkout@v2
       - uses: actions/setup-python@v2
-      - uses: pre-commit/action@v2.0.0
-  sceptre-sandbox:
-    if: github.event_name == 'push'
+      - uses: pre-commit/action@v2.0.2
+
+  sceptre-deploy:
+    name: Deploy CloudFormation templates using sceptre
     runs-on: ubuntu-latest
     needs: pre-commit
+    if: github.ref == 'refs/heads/prod'
```

Comment:
does this work with  line 7?  i'm not sure it makes sense to trigger on main & develop branch while this says only do prod branch?

Reply:
This setup is based off of what we did with for the [Nextflow accounts](https://github.com/Sage-Bionetworks-Workflows/aws-workflows-nextflow-infra/blob/prod/.github/workflows/aws-deploy.yaml). This way, the pre-commit checks are always triggered, but the deployment is reserved for pushed to `prod`.

Reply:
it's not the same @BrunoGrandePhD , your workflow trigger is set to `on: push` while this one is set to `branches: [main, develop]`

Reply:
I didn't claim it was the same. I said it was based off of the Nextflow CI workflow. 

For simplicity, I'll change the trigger to be the same so pre-commit is run on all pushes.